libpysal/graph/tests/test_spatial_lag.py

```diff
@@ -29,3 +34,34 @@ def test_lag_spatial(self):
         yl = _lag_spatial(g_row, y)
         ylc = np.array([2.0, 2.0, 3.0, 3.33333333, 4.0, 4.66666667, 5.0, 6.0, 6.0])
         np.testing.assert_array_almost_equal(yl, ylc)
+
+    def test_lag_spatial_categorical(self):
+        yl = _lag_spatial(self.gc, self.yc, categorical=True)
```

Comment:
```suggestion
        yl = _lag_spatial(self.gc, self.yc)
```
Let's leave `categorical=True` out at least once to check the automatic detection